src/Parser/Ast/UnaryOperationNode.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * PackageFactory.ComponentEngine - Universal View Components for PHP
+ *   Copyright (C) 2022 Contributors of PackageFactory.ComponentEngine
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace PackageFactory\ComponentEngine\Parser\Ast;
+
+use PackageFactory\ComponentEngine\Definition\UnaryOperator;
+use PackageFactory\ComponentEngine\Parser\Tokenizer\Scanner;
+use PackageFactory\ComponentEngine\Parser\Tokenizer\Token;
+
+final class UnaryOperationNode implements \JsonSerializable
+{
+    private function __construct(
+        public readonly UnaryOperator $operator,
+        public readonly ExpressionNode $argument
+    ) {
+    }
+
+    /**
+     * @param \Iterator<mixed,Token> $tokens
+     */
+    public static function fromTokens(\Iterator $tokens): self
+    {
+        Scanner::skipSpace($tokens);
+
+        $operator = UnaryOperator::fromTokenType(Scanner::type($tokens));
+
+        Scanner::skipOne($tokens);
+
+        $argument = ExpressionNode::fromTokens($tokens);
```

Comment:
```suggestion
        $argument = ExpressionNode::fromTokens($tokens, Precedence::UNARY);
```

`Precedence::UNARY` already exists. If applied here, the Transpiler will behave correctly.